server/api/db/sqldb/db.py

```diff
@@ -4151,28 +4152,32 @@ def _delete_tagged_object(
                 "Neither name nor key specified when deleting an object."
             )
 
-        object_id = None
         obj_name = name or key
-        if uid:
-            object_record = self._query(
-                session,
-                cls,
-                project=project,
-                uid=uid,
-                name=name,
-                key=key,
-                **kwargs,
-            ).one_or_none()
-            if object_record is None:
-                return None, None
-            object_id = object_record.id
-        elif tag and tag != "*":
-            tag_record = self._query(
-                session, cls.Tag, project=project, name=tag, obj_name=obj_name
-            ).one_or_none()
-            if tag_record is None:
-                return None, None
-            object_id = tag_record.obj_id
+
+        # try to find the object by given arguments
```

Comment:
I'm concerned since you changed the logic - before this change, if `uid` was given, then it was used to pin-point the object, regardless of tags. Now, you do this join, so it's possible that `uid` and `tag` are passed that are non compatible, resulting in no query results while before it would return the record with the given `uid`.
I'm not saying your logic is not better, I'm just thinking that it may change the results of some client logic, bad as it may be.

Reply:
We do have a validation on this specific case at the start of the function:
```python
        if tag and uid:
            raise mlrun.errors.MLRunInvalidArgumentError(
                "Both uid and tag specified when deleting an object."
            )
```

Reply:
yeah, my thought process started here but I think the main concern is the last comment which talks about a completely different scenario.